### PR TITLE
[IMP] micro_brewery: Improvement of products

### PR DESCRIPTION
**Pull request by jaeschwa**
Add new knowledge article, remove field x_beer,
and clean up studio customizations. Create new
product categories and link them to products.
Add new units of measure and update product units
accordingly. Correct base unit counts and verify
accuracy on existing SO, PO, and BOM.

task-4070521

### CODE REVIEW

File: micro_brewery/data/website_view.xml [hunk 1]
```diff
@@ -149,13 +149,6 @@
             </t>
         </field>
     </record>
-
-    <template id="micro_brewery_custom_appointment_heading" inherit_id="appointment.appointment_info">
-        <xpath expr="//h4[hasclass('pb-3')]" position="replace">
-            <h4 class="pb-3">Schedule <span style="color:#714b67;">Visit of Brewery</span></h4>
-        </xpath>
-    </template>
```
Comment (vava-odoo) [inline micro_brewery/data/website_view.xml:157]: Isn't this custo interesting? :thinking:

Comment (jaeschwa) [reply micro_brewery/data/website_view.xml:157]: Yes but it implies studio.

File: tests/test_generic/tests/test_xml.py [hunk 2]
```diff
@@ -528,3 +527,14 @@ def _check_view_active(self, xml_content, file_name):
                     record.get('id'),
                     file_name,
                 )
+    def _check_view_studio(self, xml_content, file_name):
```
[the remark this answers is not in the document]
Comment (jaeschwa) [reply tests/test_generic/tests/test_xml.py:530]: You are right. Isn't that already checked in _check_manifest ?

File: hardware_shop/__manifest__.py [hunk 3]
```diff
@@ -21,6 +21,7 @@
         'website_sale_loyalty',
         'website_sale_stock',
         'theme_kiddo',
+        'web_studio',
```
[the remark this answers is not in the document]
Comment (jaeschwa) [reply hardware_shop/__manifest__.py:24]: I think it is checked in _check_manifest.

File: micro_brewery/data/knowledge_article.xml
Comment (vava-odoo) [inline micro_brewery/data/knowledge_article.xml:1]: Sad that we can't see what changed with this commit...

File: micro_brewery/data/mrp_bom.xml [hunk 4]
```diff
@@ -31,54 +24,70 @@
         <field name="product_uom_id" ref="uom_uom_27"/>
         <field name="byproduct_ids" eval="[(6, 0, [ref('mrp_bom_byproduct_3')])]"/>
     </record>
+    <record id="mrp_bom_4" model="mrp.bom">
```
Comment (vava-odoo) [inline micro_brewery/data/mrp_bom.xml:27]: Useless diff

File: micro_brewery/data/product_product.xml [hunk 5]
```diff
@@ -149,63 +149,83 @@
...
         <field name="available_in_pos" eval="True"/>
         <field name="pos_categ_ids" eval="[(6, 0, [ref('pos_category_2')])]"/>
-        <field name="x_beer" ref="product_template_22"/>
         <field name="base_unit_id" ref="base_unit_litre"/>
+        <field name="base_unit_count">0.5</field>
+        <field name="uom_id" ref="uom_uom_30"/>
+        <field name="uom_po_id" ref="uom_uom_30"/>
         <field name="image_1920" type="base64" file="micro_brewery/static/src/binary/product_template/34-image_1920"/>
     </record>
+    <record id="product_product_29" model="product.template" context="{'create_product_product': False}">
```
Comment (vava-odoo) [inline micro_brewery/data/product_product.xml:204]: Why a product template here?

File: micro_brewery/data/uom_uom.xml [hunk 6]
```diff
@@ -6,4 +6,34 @@
...
+    <record id="uom_uom_28" model="uom.uom">
+        <field name="name">Keg</field>
+        <field name="category_id" ref="uom.product_uom_categ_vol"/>
+        <field name="factor_inv">20</field>
+        <field name="uom_type">bigger</field>
+    </record>
+    <record id="uom_uom_29" model="uom.uom">
+        <field name="name">Bottle</field>
+        <field name="category_id" ref="uom.product_uom_categ_vol"/>
+        <field name="factor_inv">0.3333333333333333333333333333</field>
```
Comment (vava-odoo) [inline micro_brewery/data/uom_uom.xml:18]:
```suggestion
        <field name="factor">3</field>
```

File: micro_brewery/data/uom_uom.xml [hunk 7]
```diff
@@ -6,4 +6,34 @@
...
+    <record id="uom_uom_31" model="uom.uom">
+        <field name="name">Half-Pint</field>
+        <field name="category_id" ref="uom.product_uom_categ_vol"/>
+        <field name="factor_inv">0.25</field>
+        <field name="uom_type">smaller</field>
+    </record>
+    <record id="uom_uom_32" model="uom.uom">
+        <field name="name">Bin</field>
+        <field name="category_id" ref="uom.product_uom_categ_vol"/>
+        <field name="factor_inv">7.92</field>
```
Comment (vava-odoo) [inline micro_brewery/data/uom_uom.xml:36]: :thinking:

**Review by vava-odoo — COMMENTED** [review]
Not much to say here. Need a functional approval.
Runbot is red :-(

File: micro_brewery/data/uom_uom.xml (hunk 7 above)
Comment (jaeschwa) [reply micro_brewery/data/uom_uom.xml:36]: What is the problem ?

File: micro_brewery/data/ir_model_data.xml [hunk 8]
```diff
@@ -6,8 +6,14 @@
...
-     <record id="cloc_exclude_appointment_heading" model="ir.model.data">
-        <field name="name">cloc_exclude_appointment_heading</field>
+     <record id="cloc_exclude_knowledge" model="ir.model.data">
+        <field name="name">cloc_exclude_knowedge_article</field>
+        <field name="model">ir.ui.view</field>
+        <field name="module">__cloc_exclude__</field>
+        <field name="res_id" eval="ref('welcome_article_body')"/>
+    </record>
+    <record id="cloc_exclude_custom_appointment" model="ir.model.data">
+        <field name="name">cloc_exclude_custom_appointment</field>
```
Comment (vava-odoo) [inline micro_brewery/data/ir_model_data.xml:16]: useless diff :innocent:

File: micro_brewery/data/product_product.xml (hunk 5 above)
Comment (vava-odoo) [reply micro_brewery/data/product_product.xml]: I assume the fix was not to move it to the template, but instead declare a product, no? (just like other 50cl beers)

**Review by vava-odoo — APPROVED** [review]
fail :innocent:

**Review by vava-odoo — APPROVED** [review]
seal-approval

**Pull request merged**